src/components/BlockInfo.vue

```diff
@@ -155,9 +158,36 @@ export default {
     toUUID,
     goToBlock: function (relativeIndex) {
       const i = parseInt(this.$route.params.blockIndex) + relativeIndex
-      if (i < 0 || i > this.blocks.length - 1) return
-      this.getBlockByIndex(i)
-      this.$router.push(`/${this.$route.params.chain}/blocks/${i}`)
+      if (i < 0) return
+      // In the case we want to fetch a block whose index is greater than the
+      // current number of loaded block, we have to check if one or more blocks
+      // have been added. To do so, we query the last stored block in the chain
+      // and compare its hash with our last local stored block. If the hash
+      // match, we can be sur that no new blocks were added. Otherwise we can
+      // safely fetch the i+1 block.
+      if (i > this.blocks.length - 1) {
+        this.socket.getLatestBlock(hex2Bytes(this.$route.params.chain), false).then(
+          (latestBlock) => {
+            if (bytes2Hex(latestBlock.hash) !== bytes2Hex(this.block.hash)) {
```

Comment:
Both .hash should be Buffers so you can use `!latestBlock.Hash.Equals(this.block.hash)` to avoid the conversion